huaweicloud/services/cloudtable/resource_huaweicloud_cloudtable_cluster.go

```diff
@@ -0,0 +1,366 @@
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z0-9-]+$`),
+						"The name can only consist of lowercase letters, numbers and hyphens (-)"),
+					validation.StringLenBetween(4, 64),
+				),
+			},
+			"storage_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"COMMON", "ULTRAHIGH",
+				}, false),
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"iam_auth_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"opentsdb_num": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"rs_num": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      2,
+				ValidateFunc: validation.IntAtLeast(2),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hbase_public_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"open_tsdb_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"opentsdb_public_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"used_storage_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"zookeeper_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCloudTableClusterParams(d *schema.ResourceData) clusters.CreateOpts {
+	result := clusters.CreateOpts{
+		Name:           d.Get("name").(string),
+		VpcId:          d.Get("vpc_id").(string),
+		StorageType:    d.Get("storage_type").(string),
+		IAMAuthEnabled: d.Get("iam_auth_enabled").(bool),
+		Datastore: clusters.Datastore{
+			Type:    "hbase",
+			Version: "1.0.6",
```

Comment:
the latest version is 1.3.1, please update it. Also we can add an attribute to show the hbase version.

Reply:
The parameter `hbase_version` has been support at **3e33e71**.